tests/unit/hook_packages/terraform/hooks/prepare/test_resource_linking.py

```diff
@@ -131,3 +132,41 @@ def test_resource_full_address_root_module(self):
         module = TFModule(None, None, {}, {}, {}, {})
         resource = TFResource("resource_address", "type", module, {})
         self.assertEqual(resource.full_address, "resource_address")
+
+    @patch("samcli.hook_packages.terraform.hooks.prepare.resource_linking._resolve_module_variable")
+    @patch("samcli.hook_packages.terraform.hooks.prepare.resource_linking._get_configuration_address")
+    def test_resolve_module_output_with_var(self, config_mock, resolve_var_mock):
+        module = TFModule("", None, {"mycoolref": "mycoolvar"}, [], {}, {"mycooloutput": References(["var.mycoolref"])})
+
+        config_mock.return_value = "mycoolref"
+
+        _resolve_module_output(module, "mycooloutput")
+
+        config_mock.assert_called_with("mycoolref")
+        resolve_var_mock.assert_called_with(module, "mycoolref")
+
+    @patch("samcli.hook_packages.terraform.hooks.prepare.resource_linking._get_configuration_address")
+    def test_resolve_module_output_with_module(self, config_mock):
+        module = TFModule("", None, {}, [], {}, {"mycooloutput": References(["module.mycoolmod"])})
+        module2 = TFModule("module.mycoolmod", module, {}, [], {}, {"mycoolmod": ConstantValue("mycoolconst")})
+        module.child_modules.update({"mycoolmod": module2})
+
+        config_mock.return_value = "mycoolmod"
+
+        results = _resolve_module_output(module, "mycooloutput")
+
+        self.assertEqual(len(results), 1)
+        self.assertEqual(results[0].value, "mycoolconst")
```

Comment:
I am not sure how this test case passed :)

Reply:
So for this test case, both the name of the module and output were the same name (I only set the reference as `module.mycoolmod` and not `module.mycoolmod.output`). Since I didn't have an output, the function managed to set both `module_name` and `output_name` to the same thing, but since the output name happened to be the same as the module name in the test, everything worked out fine and it managed to pass.

I can probably add a check in the function to make sure that we have a reference in the format `module.name.output` and throw an exception if it isn't there.

Reply:
I think we should validate this case. `module.name` is not a correct format, and we should not be able to know what is the output name.